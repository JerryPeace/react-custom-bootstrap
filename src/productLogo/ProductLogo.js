import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classNames from 'classnames';
import bootstrapUtils from '../utils/bootstrapUtils';

class ProductLogo extends Component {

    static propTypes = {
        bsClass: PropTypes.string
    };

    static defaultProps = {
        bsClass: 'banner',
        bsStyle: 'header'
    };

    render () {
        const {
            bsClass,
            children,
            className
        } = this.props;

        const bsCls = bootstrapUtils.getCls(this.props);

        return (
            <header className={classNames(className, bsCls)}>
                <h4 className={`${bsClass}-logo`}>{children}</h4>
            </header>
        );
    }
}

export default ProductLogo;
