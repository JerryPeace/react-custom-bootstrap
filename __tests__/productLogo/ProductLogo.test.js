import React from 'react';
import { mount } from 'enzyme';
import toJSON from'enzyme-to-json';
import ProductLogo from 'productLogo/ProductLogo';

describe('ProductLogo', () => {
    test('render productLogo correctly', () => {
        const productLogo = mount(<ProductLogo bsClass='ddd'/>);
        expect(toJSON(productLogo)).toMatchSnapshot();
    });
});
