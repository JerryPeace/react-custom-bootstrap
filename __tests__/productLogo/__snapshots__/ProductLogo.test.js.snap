// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductLogo render productLogo correctly 1`] = `
<ProductLogo
  bsClass="ddd"
  bsStyle="header"
>
  <header
    className=""
  >
    <h4
      className="ddd-logo"
    />
  </header>
</ProductLogo>
`;
